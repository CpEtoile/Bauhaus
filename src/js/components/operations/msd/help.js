import React, { Component } from 'react';
import MSDComponent from 'js/components/operations/msd/msd';
import { connect } from 'react-redux';
import Loading from 'js/components/shared/loading';
import { LOADING, NOT_LOADED, LOADED } from 'js/constants';
import loadMetadataStructure from 'js/actions/operations/metadatastructure/list';
import loadSIMS, { saveSims } from 'js/actions/operations/sims/item';
import { withRouter } from 'react-router-dom';
import MSDHelp from 'js/components/operations/msd/msd/visualizations/help';
import Sims from 'js/components/operations/msd/msd/visualizations/sims';
import SimsForm from 'js/components/operations/msd/msd/visualizations/sims-form';
import buildExtract from 'js/utils/build-extract';
import PropTypes from 'prop-types';
import { saveSecondLang } from 'js/actions/app';
import * as select from 'js/reducers';

const extractId = buildExtract('id');
const extractIdOperation = buildExtract('idOperation');

export const HELP = 'HELP';
export const CREATE = 'CREATE';
export const VIEW = 'VIEW';

class MSDContainer extends Component {
	componentWillMount() {
		if (this.props.status !== LOADED) {
			this.props.loadMetadataStructure();
			this.props.mode === VIEW && this.props.loadSIMS(this.props.id);
		}
	}
	componentWillReceiveProps(nextProps) {
		// If we do a redirect form the Edit to the view mode, we must reload the sims data from the server
		if (
			nextProps.mode === VIEW &&
			nextProps.id !== this.props.id &&
			nextProps.id
		) {
			this.props.loadSIMS(nextProps.id);
		}
	}
	render() {
		const {
			metadataStructure,
			status,
			codesLists,
			mode = HELP,
			baseUrl,
			saveSims,
			idOperation,
			disableSectionAnchor,
			saveSecondLang,
			langs,
			secondLang,
			currentSims,
		} = this.props;
		if (status !== LOADED)
			return <Loading textType="loading" context="operations" />;
		if (mode === VIEW && !currentSims.id)
			return <Loading textType="loading" context="operations" />;
		return (
			<MSDComponent
				metadataStructure={metadataStructure}
				currentSection={this.props.match.params.idSection}
				storeCollapseState={mode === HELP}
				baseUrl={baseUrl}
				disableSectionAnchor={disableSectionAnchor}
			>
				{mode === HELP && (
					<MSDHelp
						metadataStructure={metadataStructure}
						codesLists={codesLists}
						currentSection={this.props.match.params.idSection}
					/>
				)}

				{mode === VIEW && (
					<Sims
						sims={currentSims.rubrics}
						metadataStructure={metadataStructure}
						codesLists={codesLists}
						currentSection={this.props.match.params.idSection}
						saveSecondLang={saveSecondLang}
						langs={langs}
						secondLang={secondLang}
					/>
				)}
				{mode === CREATE && (
					<SimsForm
						sims={currentSims.rubrics}
						metadataStructure={metadataStructure}
						codesLists={codesLists}
						currentSection={this.props.match.params.idSection}
						onSubmit={saveSims}
						idOperation={idOperation}
						saveSecondLang={saveSecondLang}
						langs={langs}
						secondLang={secondLang}
					/>
				)}
			</MSDComponent>
		);
	}
}

const mapStateToProps = (state, ownProps) => {
	if (!state.operationsMetadataStructureList) {
		return {
			status: NOT_LOADED,
			metadataStructure: [],
		};
	}
	const {
		results: metadataStructure,
		status,
		err,
	} = state.operationsMetadataStructureList;

	return {
		langs: select.getLangs(state),
		secondLang: state.app.secondLang,
		metadataStructure,
		currentSims: ownProps.mode === VIEW ? state.operationsSimsCurrent : {},
		id: extractId(ownProps),
		idOperation: extractIdOperation(ownProps),
		codesLists: state.operationsCodesList.results,
		status,
		err,
	};
};

const mapDispatchToProps = {
	loadMetadataStructure,
	loadSIMS,
	saveSims,
	saveSecondLang,
};

export default withRouter(
	connect(
		mapStateToProps,
		mapDispatchToProps
	)(MSDContainer)
);

MSDContainer.propTypes = {
	metadataStructure: PropTypes.object,
	status: PropTypes.oneOf([LOADED, NOT_LOADED, LOADING]),
	codesLists: PropTypes.object,
	mode: PropTypes.oneOf([HELP, VIEW, CREATE]),
	baseUrl: PropTypes.string,
	disableSectionAnchor: PropTypes.bool,
};
MSDContainer.defaultProps = {
	currentSims: {},
};
