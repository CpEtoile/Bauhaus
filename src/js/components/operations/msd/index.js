import React, { Component } from 'react';
import MSDLayout from 'js/components/operations/msd/layout/';
import { connect } from 'react-redux';
import Loading from 'js/components/shared/loading';
import { LOADING, NOT_LOADED, LOADED } from 'js/constants';
import loadMetadataStructure from 'js/actions/operations/metadatastructure/list';
import loadSIMS, { saveSims } from 'js/actions/operations/sims/item';
import { withRouter } from 'react-router-dom';
import MSDHelp from 'js/components/operations/msd/pages/help';
import SimsVisualisation from 'js/components/operations/msd/pages/sims-visualisation/';
import SimsCreation from 'js/components/operations/msd/pages/sims-creation/';
import buildExtract from 'js/utils/build-extract';
import PropTypes from 'prop-types';
import { saveSecondLang } from 'js/actions/app';
import { compose } from 'recompose';
import * as select from 'js/reducers';
import PageSubtitle from 'js/components/shared/page-subtitle';
import PageTitle from 'js/components/shared/page-title';
import loadOperation from 'js/actions/operations/operations/item';
import D from 'js/i18n';
import { getMessageForSecondLang } from 'js/i18n/build-dictionary';
import loadOperationsList from 'js/actions/operations/operations/list';

const labelOperationNameTemplate = '{{OPERATION_LABEL}}';

const extractId = buildExtract('id');
const extractIdOperation = buildExtract('idOperation');

export const HELP = 'HELP';
export const CREATE = 'CREATE';
export const VIEW = 'VIEW';
export const UPDATE = 'UPDATE';
export const DUPLICATE = 'DUPLICATE';

class MSDContainer extends Component {
	static propTypes = {
		metadataStructure: PropTypes.object,
		metadataStructureStatus: PropTypes.oneOf([LOADED, NOT_LOADED, LOADING]),
		codesLists: PropTypes.object,
		mode: PropTypes.oneOf([HELP, VIEW, CREATE, UPDATE, DUPLICATE]),
		baseUrl: PropTypes.string,
		id: PropTypes.string,
		saveSims: PropTypes.func,
		idOperation: PropTypes.string,
		disableSectionAnchor: PropTypes.bool,
		saveSecondLang: PropTypes.func,
		langs: PropTypes.object,
		secondLang: PropTypes.bool,
		currentSims: PropTypes.object,
	};
	static defaultProps = {
		currentSims: {},
	};

	constructor(props) {
		super(props);
		this.goBackCallback = this.goBackCallback.bind(this);
	}

	goBackCallback(url) {
		this.props.history.push(url);
	}

	componentWillMount() {
		if (this.props.metadataStructureStatus !== LOADED) {
			this.props.loadMetadataStructure();
		}
		if (this.props.operationsListStatus !== LOADED) {
			this.props.loadOperationsList();
		}
		if (!this.props.currentSims.id) {
			this.props.loadSIMS(this.props.id);
		}
		if (!this.props.isOperationLoaded) {
			this.props.loadOperation(this.props.idOperation);
		}
	}

	componentWillReceiveProps(nextProps) {
		if (!nextProps.currentSims.id || this.props.id !== nextProps.id) {
			this.props.loadSIMS(nextProps.id);
		}
		if (!nextProps.isOperationLoaded) {
			this.props.loadOperation(nextProps.idOperation);
		}
	}
	render() {
		const {
			metadataStructure,
			metadataStructureStatus,
			codesLists,
			mode = HELP,
			baseUrl,
			saveSims,
			idOperation,
			disableSectionAnchor,
			saveSecondLang,
			langs,
			secondLang,
			currentSims,
			operations = [],
		} = this.props;

		if (
			metadataStructureStatus !== LOADED ||
			(mode === VIEW && !currentSims.id)
		)
			return <Loading textType="loading" context="operations" />;

		return (
			<MSDLayout
				metadataStructure={metadataStructure}
				currentSection={this.props.match.params.idSection}
				storeCollapseState={mode === HELP}
				baseUrl={baseUrl}
				disableSectionAnchor={disableSectionAnchor}
			>
				{mode !== HELP && mode !== DUPLICATE && (
					<React.Fragment>
						<PageTitle title={currentSims.labelLg1} context="operations" />
						{secondLang && (
							<PageSubtitle
								subTitle={currentSims.labelLg2}
								context="operations"
							/>
						)}
					</React.Fragment>
				)}
				{mode === HELP && (
					<MSDHelp
						metadataStructure={metadataStructure}
						codesLists={codesLists}
						currentSection={this.props.match.params.idSection}
					/>
				)}

				{mode === VIEW && (
					<SimsVisualisation
						sims={currentSims}
						idOperation={currentSims.idOperation}
						metadataStructure={metadataStructure}
						codesLists={codesLists}
						currentSection={this.props.match.params.idSection}
						saveSecondLang={saveSecondLang}
						langs={langs}
						secondLang={secondLang}
						goBack={this.goBackCallback}
					/>
				)}
				{(mode === CREATE || mode === UPDATE || mode === DUPLICATE) && (
					<SimsCreation
						sims={currentSims}
						metadataStructure={metadataStructure}
						codesLists={codesLists}
						onSubmit={saveSims}
						idOperation={idOperation}
						saveSecondLang={saveSecondLang}
						langs={langs}
						secondLang={secondLang}
						goBack={this.goBackCallback}
						mode={mode}
						operations={operations}
					/>
				)}
			</MSDLayout>
		);
	}
}

const mapStateToProps = (state, ownProps) => {
	if (!state.operationsMetadataStructureList) {
		return {
			metadataStructureStatus: NOT_LOADED,
			metadataStructure: [],
		};
	}

	const {
		results: metadataStructure,
		status: metadataStructureStatus,
	} = state.operationsMetadataStructureList;

	const id = extractId(ownProps);

	const currentOperation = select.getOperation(state);
	const idOperation = extractIdOperation(ownProps);

	let currentSims = {};

	switch (ownProps.mode) {
		case HELP:
			currentSims = {};
			break;
		case CREATE:
			currentSims = {
				labelLg1: D.simsLabel.replace(
					labelOperationNameTemplate,
					currentOperation.prefLabelLg1
				),
				labelLg2: getMessageForSecondLang('simsLabel').replace(
					labelOperationNameTemplate,
					currentOperation.prefLabelLg2
				),
			};
			break;
		default:
			currentSims = select.getOperationsSimsCurrent(state);
			break;
	}

	const {
		results: operations,
		operationsListStatus,
	} = state.operationsOperationsList;

	return {
		langs: select.getLangs(state),
		secondLang: state.app.secondLang,
		metadataStructure,
		metadataStructureStatus,

		currentSims: !id || currentSims.id === id ? currentSims : {},
		isOperationLoaded:
			ownProps.mode !== CREATE || currentOperation.id === idOperation,
		id,
		idOperation,
		codesLists: state.operationsCodesList.results,
		operations,
		operationsListStatus,
	};
};

const mapDispatchToProps = {
	loadMetadataStructure,
	loadSIMS,
	saveSims,
	loadOperation,
	saveSecondLang,
	loadOperationsList,
};

export default compose(
	withRouter,
	connect(
		mapStateToProps,
		mapDispatchToProps
	)
)(MSDContainer);
