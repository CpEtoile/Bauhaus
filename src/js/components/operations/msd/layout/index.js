import React, { Component } from 'react';
import Outline from 'js/components/operations/msd/outline/';
import D from 'js/i18n';
import PropTypes from 'prop-types';

import './style.scss';

const STATUS = {
	BOTH: 'BOTH',
	SUMMARY: 'SUMMARY',
	CONTENT: 'CONTENT',
};
class MSDComponent extends Component {
	static propTypes = {
		metadataStructure: PropTypes.object.isRequired,
		storeCollapseState: PropTypes.bool,
		children: PropTypes.array,
		baseUrl: PropTypes.string,
		disableSectionAnchor: PropTypes.bool,
	};

	constructor(props) {
		super(props);
		this.state = {
			status: localStorage.getItem('HELP_VIEW') || STATUS.BOTH,
		};
	}

	changeStatus(status) {
		localStorage.setItem('HELP_VIEW', status);
		this.setState(previousState => ({
			status,
		}));
	}
	render() {
		const { status } = this.state;

		const styleSummary = {
			width: status === STATUS.BOTH ? '33%' : '100%',
			display: status === STATUS.CONTENT ? 'none' : 'block',
		};
		const styleContent = {
			width: status === STATUS.BOTH ? '66%' : '100%',
			display: status === STATUS.SUMMARY ? 'none' : 'block',
		};
		const {
			storeCollapseState,
			metadataStructure,
			children,
			baseUrl,
			disableSectionAnchor,
		} = this.props;
		return (
			<div
				id="consulter-sommaire"
				className="container panneau produit-sommaire"
			>
				<section className="sommaire-gauche" style={styleSummary}>
					<div className="titre-sommaire titre">{D.helpSummary}</div>
					<input
						className="form-control"
						disabled
						placeholder={D.search}
						aria-label={D.search}
					/>
					<nav className="sommaire-container">
						<ul className="sommaire">
							{Object.values(metadataStructure).map(metadata => (
								<Outline
									key={metadata.idMas}
									storeCollapseState={storeCollapseState}
									metadataStructure={metadata}
									baseUrl={baseUrl}
									disableSectionAnchor={disableSectionAnchor}
								/>
							))}
						</ul>
					</nav>
				</section>

				{status === STATUS.CONTENT && (
					<button
						className="ouverture-sommaire"
						onClick={() => this.changeStatus(STATUS.BOTH)}
					>
						{D.helpSummary}
						<span className="glyphicon glyphicon-chevron-right" />
					</button>
				)}
				{status === STATUS.BOTH && (
					<div className="fermeture">
						<div className="fermeture-gauche">
							<button
								onClick={() => this.changeStatus(STATUS.CONTENT)}
								title="open summary"
							>
								<span className="glyphicon glyphicon-chevron-left" />
							</button>
						</div>
						<div className="fermeture-droite">
							<button
								onClick={() => this.changeStatus(STATUS.SUMMARY)}
								title="open content"
							>
								<span className="glyphicon glyphicon-chevron-right" />
							</button>
						</div>
					</div>
				)}
				{status === STATUS.SUMMARY && (
					<button
						className="ouverture-contenu"
						onClick={() => this.changeStatus(STATUS.BOTH)}
					>
						<span className="glyphicon glyphicon-chevron-left" />
						{D.helpContent}
					</button>
				)}
				<section
					style={styleContent}
					className={
						status === STATUS.CONTENT
							? 'sommaire-droite alone'
							: 'sommaire-droite'
					}
				>
					{children}
				</section>
			</div>
		);
	}
}

export default MSDComponent;
