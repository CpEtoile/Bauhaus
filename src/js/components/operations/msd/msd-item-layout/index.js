import React from 'react';
import PropTypes from 'prop-types';

export default function MSDItemLayout({ title, children }) {
	return (
		<article className="panel panel-default contenu">
			<div className="panel-heading">
				<h3>{title}</h3>
			</div>
			<div className="panel-body">{children}</div>
		</article>
	);
}

MSDItemLayout.propTypes = {
	title: PropTypes.string.isRequired,
	children: PropTypes.object.isRequired,
};
