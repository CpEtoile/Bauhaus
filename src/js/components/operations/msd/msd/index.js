import React, { Component } from 'react';
import SummaryItem from 'js/components/operations/msd/msd/item';
import D from 'js/i18n';
import PropTypes from 'prop-types';

import './style.css';

const STATUS = {
	BOTH: 'BOTH',
	SUMMARY: 'SUMMARY',
	CONTENT: 'CONTENT',
};
class MSDComponent extends Component {
	constructor(props) {
		super(props);
		this.state = {
			status: localStorage.getItem('HELP_VIEW') || STATUS.BOTH,
		};
	}

	changeStatus(status) {
		localStorage.setItem('HELP_VIEW', status);
		this.setState({
			status,
		});
	}
	render() {
		const { status } = this.state;

		const styleSummary = {
			width: status === STATUS.BOTH ? '33%' : '100%',
			display: status === STATUS.CONTENT ? 'none' : 'block',
		};
		const styleContent = {
			width: status === STATUS.BOTH ? '66%' : '100%',
			display: status === STATUS.SUMMARY ? 'none' : 'block',
		};
		const { storeCollapseState, metadataStructure, children } = this.props;
		return (
			<div
				id="consulter-sommaire"
				className="container panneau produit-sommaire"
			>
				<section className="sommaire-gauche" style={styleSummary}>
					<div className="titre-sommaire titre">{D.helpSummary}</div>
					<input
						className="form-control"
						disabled
						placeholder={D.search}
						aria-label={D.search}
					/>
					<nav className="sommaire-container">
						<ul className="sommaire">
							{Object.keys(metadataStructure).map(id => (
								<SummaryItem
									key={id}
									storeCollapseState={storeCollapseState}
									metadataStructure={metadataStructure[id]}
								/>
							))}
						</ul>
					</nav>
				</section>

				{status === STATUS.CONTENT && (
					<button
						className="ouverture-sommaire"
						onClick={() => this.changeStatus(STATUS.BOTH)}
					>
						{D.helpSummary}
						<span className="glyphicon glyphicon-chevron-right" />
					</button>
				)}
				{status === STATUS.BOTH && (
					<div className="fermeture">
						<div className="fermeture-gauche">
							<button
								onClick={() => this.changeStatus(STATUS.CONTENT)}
								title="open summary"
							>
								<span className="glyphicon glyphicon-chevron-left" />
							</button>
						</div>
						<div className="fermeture-droite">
							<button
								onClick={() => this.changeStatus(STATUS.SUMMARY)}
								title="open content"
							>
								<span className="glyphicon glyphicon-chevron-right" />
							</button>
						</div>
					</div>
				)}
				{status === STATUS.SUMMARY && (
					<button
						className="ouverture-contenu"
						onClick={() => this.changeStatus(STATUS.BOTH)}
					>
						<span className="glyphicon glyphicon-chevron-left" />
						{D.helpContent}
					</button>
				)}
				<section
					style={styleContent}
					className={
						status === STATUS.CONTENT
							? 'sommaire-droite alone'
							: 'sommaire-droite'
					}
				>
					{children}
				</section>
			</div>
		);
	}
}

MSDComponent.propTypes = {
	metadataStructure: PropTypes.object.isRequired,
	storeCollapseState: PropTypes.bool,
};

export default MSDComponent;
