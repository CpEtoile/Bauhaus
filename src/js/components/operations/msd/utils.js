export function isOpen(id) {
	const collapsed = JSON.parse(localStorage.getItem('HELP_COLLAPSED')) || {};
	return collapsed[id] || false;
}

export function toggleOpen(id) {
	const collapsed = JSON.parse(localStorage.getItem('HELP_COLLAPSED')) || {};
	const previous = collapsed[id] || false;
	localStorage.setItem(
		'HELP_COLLAPSED',
		JSON.stringify({
			...collapsed,
			[id]: !previous,
		})
	);
}
