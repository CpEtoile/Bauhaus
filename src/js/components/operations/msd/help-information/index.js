import React from 'react';
import D from 'js/i18n';
import { rangeType } from 'js/utils/msd/';
const { CODE_LIST } = rangeType;

export default function HelpInformation({ msd, codesLists }) {
	if (!msd.masLabelLg1) {
		return null;
	}
	return (
		<dl>
			<dt>{D.labelTitle}:</dt>
			<dd>{msd.masLabelLg2}</dd>
			<dt>{D.helpPresentational}:</dt>
			<dd>{msd.isPresentational.toString()}</dd>
			<dt>{D.helpRange}:</dt>
			<dd>
				{msd.rangeType === CODE_LIST && codesLists[msd.codeList]
					? `${D[`help${msd.rangeType}`]} - ${
							codesLists[msd.codeList].codeListLabelLg1
					  }`
					: `${D[`help${msd.rangeType}`]}`}

				{msd.rangeType === CODE_LIST &&
					codesLists[msd.codeList] && (
						<ul className="list-group">
							{codesLists[msd.codeList].codes.map(code => (
								<li className="list-group-item" key={code.code}>
									{code.labelLg1}
								</li>
							))}
						</ul>
					)}
			</dd>
		</dl>
	);
}
