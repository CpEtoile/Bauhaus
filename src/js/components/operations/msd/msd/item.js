import React, { Component } from 'react';
import { toggleOpen, isOpen } from 'js/components/operations/msd/msd/utils';
import { HashLink as Link } from 'react-router-hash-link';
import SectionItem from 'js/components/operations/msd/msd/section-item';
import PropTypes from 'prop-types';

class SummaryItem extends Component {
	constructor(props) {
		super(props);

		/**
		 * If the storeCollapseState is false, we won't store the state in the localStorage.
		 * It will be resetted after each F5
		 */
		this.state = {
			opened: props.storeCollapseState && isOpen(props.metadataStructure.idMas),
		};
	}

	expandOrCollapseItem() {
		this.setState({ opened: !this.state.opened });
		this.props.storeCollapseState &&
			toggleOpen(this.props.metadataStructure.idMas);
	}

	render() {
		const {
			metadataStructure,
			storeCollapseState,
			baseUrl = '/operations/help/',
			disableSectionAnchor = false,
		} = this.props;

		return (
			<li>
				<div className="primary-item">
					<Link
						to={`${baseUrl}${
							disableSectionAnchor ? '' : metadataStructure.idMas
						}#${metadataStructure.idMas}`}
					>
						{metadataStructure.idMas} - {metadataStructure.masLabelLg1}
					</Link>
					<button
						className="up-down"
						title={metadataStructure.masLabelLg1}
						onClick={() => this.expandOrCollapseItem()}
					>
						{Object.keys(metadataStructure.children).length > 0 && (
							<span
								className={` glyphicon glyphicon-chevron-${
									this.state.opened ? 'up' : 'down'
								}`}
							/>
						)}
					</button>
				</div>

				{this.state.opened && (
					<SectionItem
						children={metadataStructure.children}
						parent={metadataStructure.idMas}
						storeCollapseState={storeCollapseState}
						baseUrl={baseUrl}
						disableSectionAnchor={disableSectionAnchor}
					/>
				)}
			</li>
		);
	}
}

SummaryItem.propTypes = {
	storeCollapseState: PropTypes.bool,
	metadataStructure: PropTypes.object.isRequired,
	baseUrl: PropTypes.string,
};

export default SummaryItem;
