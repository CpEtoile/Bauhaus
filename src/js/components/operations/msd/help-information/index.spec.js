import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import HelpInformation from './';
import { rangeType } from 'js/utils/msd/';
const { CODE_LIST, TEXT } = rangeType;

describe('HelpInformation', () => {
	it('should return null if the masLabelLg1 is undefined', () => {
		const instance = shallow(<HelpInformation msd={{}} />);
		expect(instance.html()).toBeNull();
	});

	it('should display the right content when the rangeType is a CODE_LIST', () => {
		const msd = {
			masLabelLg1: 'masLabelLg1',
			masLabelLg2: 'masLabelLg2',
			isPresentational: true,
			rangeType: CODE_LIST,
			codeList: 'codeList',
		};
		const codesLists = {
			codeList: {
				codeListLabelLg1: 'codeListLabelLg1',
				codes: [{ code: 'code1' }, { code: 'code2' }],
			},
		};
		const tree = renderer
			.create(<HelpInformation msd={msd} codesLists={codesLists} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should display the right content when the rangeType is not a CODE_LIST', () => {
		const msd = {
			masLabelLg1: 'masLabelLg1',
			masLabelLg2: 'masLabelLg2',
			isPresentational: true,
			rangeType: TEXT,
		};
		const tree = renderer.create(<HelpInformation msd={msd} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
