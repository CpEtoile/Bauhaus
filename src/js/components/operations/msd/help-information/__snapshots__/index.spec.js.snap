// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpInformation should display the right content when the rangeType is a CODE_LIST 1`] = `
<dl>
  <dt>
    Label
    :
  </dt>
  <dd>
    masLabelLg2
  </dd>
  <dt>
    Presentational attribut
    :
  </dt>
  <dd>
    true
  </dd>
  <dt>
    Range
    :
  </dt>
  <dd>
    Code list - codeListLabelLg1
    <ul
      className="list-group"
    >
      <li
        className="list-group-item"
      />
      <li
        className="list-group-item"
      />
    </ul>
  </dd>
</dl>
`;

exports[`HelpInformation should display the right content when the rangeType is not a CODE_LIST 1`] = `
<dl>
  <dt>
    Label
    :
  </dt>
  <dd>
    masLabelLg2
  </dd>
  <dt>
    Presentational attribut
    :
  </dt>
  <dd>
    true
  </dd>
  <dt>
    Range
    :
  </dt>
  <dd>
    Text
  </dd>
</dl>
`;
